lib/Controller/UsersController.php

```diff
@@ -469,6 +501,254 @@ public function create($username, $password, array $groups= [], $email='') {
+				'core', 'error',
+				[
+					"errors" => [["error" => $this->l10n->t('User: %s does not exist', [$userId])]]
+				],
+				'guest'
+			);
+		}
+
+		if ($user->getEMailAddress() === null) {
+			return new TemplateResponse(
+				'core', 'error',
+				[
+					"errors" => [["error" => $this->l10n->t('Email address not set for: %s', [$userId])]]
+				],
+				'guest'
+			);
+		}
+
+		$token = $this->secureRandom->generate(21,
+			ISecureRandom::CHAR_DIGITS,
+			ISecureRandom::CHAR_LOWER, ISecureRandom::CHAR_UPPER);
+		$this->config->setUserValue($userId, 'owncloud',
+			'lostpassword', $this->timeFactory->getTime() . ':' . $token);
+
+		// data for the mail template
+		$mailData = [
+			'username' => $userId,
+			'url' => $this->urlGenerator->linkToRouteAbsolute('user_management.Users.setPasswordForm', ['userId' => $userId, 'token' => $token])
+		];
+
+		$mail = new TemplateResponse('user_management', 'new_user/email-html', $mailData, 'blank');
+		$mailContent = $mail->render();
+
+		$mail = new TemplateResponse('user_management', 'new_user/email-plain_text', $mailData, 'blank');
+		$plainTextMailContent = $mail->render();
+
+		$subject = $this->l10n->t('Your %s account was created', [$this->defaults->getName()]);
+
+		try {
+			$message = $this->mailer->createMessage();
+			$message->setTo([$user->getEMailAddress() => $userId]);
+			$message->setSubject($subject);
+			$message->setHtmlBody($mailContent);
+			$message->setPlainBody($plainTextMailContent);
+			$message->setFrom([$this->fromMailAddress => $this->defaults->getName()]);
+			$this->mailer->send($message);
+		} catch (\Exception $e) {
+			$this->log->error("Can't send new user mail to " . $user->getEMailAddress() . ": " . $e->getMessage(), ['app' => 'settings']);
+			return new TemplateResponse(
+				'core', 'error',
+				[
+					"errors" => [[
+						"error" => $this->l10n->t('Can\'t send email to %s for the new user. ' .
+							'Contact your administrator.', [$user->getEMailAddress()])]]
+				], 'guest'
+			);
+		}
+
+		return new TemplateResponse(
+			'user_management', 'new_user/tokensendnotify', [], 'guest'
+		);
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param $token
+	 * @param $userId
+	 * @param $password
+	 * @return JSONResponse
+	 */
+	public function setPassword($token, $userId, $password) {
+		$user = $this->userManager->get($userId);
+
+		if ($user === null) {
+			return new JSONResponse(
+				[
+					'status' => 'error',
+					'message' => $this->l10n->t('User: %s does not exist. ' .
+						'Contact the administrator.', [$userId]),
+					'type' => 'usererror'
+				], Http::STATUS_NOT_FOUND
+			);
+		}
+
+		try {
+			$this->checkPasswordSetToken($token, $userId);
+
+			if (!$user->setPassword($password)) {
+				return new JSONResponse(
+					[
+						'status' => 'error',
+						'message' => $this->l10n->t('The password can not be set for user: %s. ' .
+							'Contact your administrator.', [$userId]),
```

Comment:
not sure if the l10n parser bot is able to understand wrapped strings. try putting the full string on a single line instead of concatenating

Reply:
Done.